### PR TITLE
Add changes to make plugin work on ChartJS V2 & V3

### PR DESCRIPTION
**Pull request by devotox**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -19,29 +19,58 @@ function addColors(gradient, scale, colors) {
   }
 }
 
-const areaIsValid = (area) => area && area.right > area.left && area.bottom > area.top;
+function getScale(meta, axis) {
+  const key = axis + 'Scale'; 
+
+  if(meta.controller) {
```
Comment (kurkle) [inline src/index.js:25]: Both, v2 and v3 have `meta.controller`, so this in not a valid way to separate the versions.

`Chart.version` was introduced in v3, so could use that to separate.

File: src/index.js [hunk 2]
```diff
@@ -19,29 +19,58 @@ function addColors(gradient, scale, colors) {
...
+
+  if(meta.dataset.options) {
+    meta.dataset.options[key] = value;
+  } else {
+    meta.dataset[key] = value;
+  }
+}
+
+
+function areaIsValid(area){
```
Comment (kurkle) [inline src/index.js:43]: Why would this need to be changed?

File: src/index.js [hunk 3]
```diff
@@ -19,29 +19,58 @@ function addColors(gradient, scale, colors) {
...
+  return area && area.right > area.left && area.bottom > area.top;
+}
 
 export default {
   id: 'gradient',
   beforeDatasetsUpdate(chart) {
     const area = chart.chartArea;
+
```
Comment (kurkle) [inline src/index.js:51]: No need to re-style the code

**Review by kurkle — CHANGES_REQUESTED** [review]

**Comment by devotox** [comment]
@kurkle Thanks for the review changes have been made

File: src/index.js [hunk 4]
```diff
@@ -19,6 +19,26 @@ function addColors(gradient, scale, colors) {
   }
 }
 
+function getScale(chart, meta, axis) {
+  const key = axis + 'Scale'; 
+
+  if(chart.version) {
```
Comment (kurkle) [inline src/index.js:25]: The version is in the constructor, not in the instance.
Could be for example: 
```js
const getScale = Chart.version 
  ? (meta, axis) => meta[axis + 'Scale'] 
  : (meta, axis) => meta.controller['_' + key + 'Scale'];
```

**Review by kurkle — CHANGES_REQUESTED** [review]

**Review by kurkle — APPROVED** [review]

**Pull request merged**